### PR TITLE
fix(replays): reconstruct selector queries

### PR DESCRIPTION
**Pull request by michellewzhang**
`dead.selector` and `rage.selector` queries are now more accurate.

Before: search was returning too many replays

SCR-20231004-sged

After: search is returning the correct replays

SCR-20231004-sfon

Loading placeholders:
Screenshot 2023-10-05 at 12 50 35 PM
Screenshot 2023-10-05 at 12 50 30 PM

https://github.com/getsentry/sentry/assets/56095982/11977b8e-afd3-4e8a-a5a5-51e0abcae1b1

### CODE REVIEW

File: static/app/views/replays/deadRageClick/selectorTable.tsx [hunk 1]
```diff
@@ -32,6 +32,41 @@ export function getAriaLabel(str: string) {
   return pre.substring(0, pre.lastIndexOf('"]'));
 }
 
+function trimAttribute(elementAttribute, fullAlltribute) {
+  return elementAttribute === '' ? '' : fullAlltribute;
+}
+
+function constructSelector(element: Element) {
```
Comment (billyvg) [inline static/app/views/replays/deadRageClick/selectorTable.tsx:39]: Are there any tests for this?

Comment (michellewzhang) [reply static/app/views/replays/deadRageClick/selectorTable.tsx:39]: will write one!

File: static/app/views/replays/types.tsx [hunk 2]
```diff
@@ -181,6 +181,17 @@ export type DeadRageSelectorListResponse = {
   data: {count_dead_clicks: number; count_rage_clicks: number; dom_element: string}[];
 };
 
+export type Element = {
```
Comment (ryan953) [inline static/app/views/replays/types.tsx:184]: Nit: there's a lot of things called Element, maybe `ReplayClickElement`?

File: static/app/views/replays/types.tsx [hunk 3]
```diff
@@ -170,7 +170,7 @@ export interface ReplayError {
 
 export type DeadRageSelectorItem = {
   aria_label: string;
-  dom_element: string;
+  dom_element: {fullSelector: string; projectId: number; selector: string};
   element: string;
```
Comment (ryan953) [inline static/app/views/replays/types.tsx:174]: should this be `string` still?
```suggestion
  element: ReplayClickElement;
```

or just `element: Element`

Comment (ryan953) [reply static/app/views/replays/types.tsx:174]: oh, sry nvm

File: static/app/views/replays/types.tsx [hunk 4]
```diff
@@ -181,6 +181,17 @@ export type DeadRageSelectorListResponse = {
   data: {count_dead_clicks: number; count_rage_clicks: number; dom_element: string}[];
```
Comment (ryan953) [inline static/app/views/replays/types.tsx:181]: does this type also include `element: ReplayClickElement` in there?

Comment (michellewzhang) [reply static/app/views/replays/types.tsx:181]: yes!

Comment (michellewzhang) [reply static/app/views/replays/types.tsx:181]: forgot to update this type along the way

File: static/app/views/replays/types.tsx (hunk 3 above)
Comment (michellewzhang) [reply static/app/views/replays/types.tsx:174]: we don't even need `element` i totally forgot to remove it

**Review by ryan953 — APPROVED** [review]

**Pull request merged**